File: app/src/main/java/com/coinninja/coinkeeper/model/db/converter/BTCStateConverter.java
package com.coinninja.coinkeeper.model.db.converter;

import com.coinninja.coinkeeper.model.db.enums.BTCState;

import org.greenrobot.greendao.converter.PropertyConverter;

public class BTCStateConverter implements PropertyConverter<BTCState, Integer> {

    @Override
    public BTCState convertToEntityProperty(Integer databaseValue) {
        if (databaseValue == null) {
            return null;
        }

        for (BTCState state : BTCState.values()) {
            if (state.getId() == databaseValue) {
                return state;
            }
        }

        return BTCState.UNFULFILLED;
    }

    @Override
    public Integer convertToDatabaseValue(BTCState state) {
        return state == null ? null : state.getId();
    }
}

File: persistance/src/main/java/app/coinninja/cn/persistance/converter/BTCStateConverter.kt
package app.coinninja.cn.persistance.converter

import androidx.room.TypeConverter
import app.coinninja.cn.persistance.model.BTCState

object BTCStateConverter {

    @TypeConverter
    @JvmStatic
    fun to(id: Int?): BTCState? = id?.let { return BTCState.from(it) }

    @TypeConverter
    @JvmStatic
    fun from(state: BTCState?): Int? = state?.let { return it.id }

}
